docs/api/connectivity/bluetooth/Optimising_for_performance.md

```diff
@@ -0,0 +1,173 @@
+# Optimising application for throughput and power consumption
+
+BLE devices are usually battery powered so performance might mean different things in different applications. You have
+to decide what is more important in yours - minimising power consumption on one or both sides of the communication or
+maximising throughput and/or latency. Some optimisation steps can in fact achieve both.
+
+This guide will discuss some trade-offs that should be considered and best practices that improve performance on all
+fronts.
+
+## Power consumtpion
+
+Any radio activity will consume power. Depending on what the stack is doing you have to power the radio even when no
+data is being sent. It is important to understand when radio is active.
+
+### Connections
+
+The most intuitive power consumption rate to understand is when using connections. Each take turns sending and receiving
+packets at set interval.
+
+
+    CENTRAL
+    ┌────┐ ┌────┐           ┌────┐ ┌────┐         ┌────┐ ┌────┐ ┌────┐ ┌────┐
+    │send│ │recv│           │send│ │recv│         │send│ │recv│ │send│ │recv│
+    └────┘ └────┘           └────┘ └────┘         └────┘ └────┘ └────┘ └────┘
+      connection interval
+     ◄─────────────────────►
+
+    PERIPHERAL
+    ┌────┐ ┌────┐                                 ┌────┐ ┌────┐ ┌────┐ ┌────┐
+    │recv│ │send│                                 │recv│ │send│ │recv│ │send│
+    └────┘ └────┘                                 └────┘ └────┘ └────┘ └────┘
+      slave latency
+     ◄───────────────────────────────────────────►
+
+    ▲                       ▲                     ▲
+    connection event        connection event      connection event
+
+
+To maintain a connection, regardless if there is data transfer to be transferred, the central needs to transmit and
+receive once very connection interval.
+
+The peripheral needs to receive and may transmit data if it has any. It may skip a number of these connection events
+set by `slaveLatency`. If it has no data to transmit empty packets are sent.
+
+More power is consumed if there is data to be exchanged. The exchange can continue until the next connection event would
+take place.
+
+It's worth considering if keeping the connection active is worth it. Connection in BLE is extremely fast and if you plan
```

Comment:
That's partially true, to save power with advertising, the advertising interval must be stretched a lot and this increase the connection latency. Maintaining a connection with an interval of 1s cost much less compared to advertising every second as it isn't sending packets on 3 channels and advertising packets are usual larger than empty. 

If the peripheral is advertising intermittently this also report the power cost to the central which must always listen.

Max advertising interval: 10.24s
Max supervision timeout: 32s

It is possible to set a connection interval of 1 second with a slave latency of 15 to be in acceptable range for the  supervision timeout. The slave would listen and ack every 15s if it has nothing to transmit.  

So maybe mention than choosing to maintain the connection active depends on the connection latency expected and power budget of the central too. Below it is mentioned.

Reply:
rephrased it to show the potential in saving for one side